src/test/kotlin/org/taktik/icure/asynclogic/impl/filter/maintenancetask/MaintenanceTaskAfterDateFilterTest.kt

```diff
@@ -0,0 +1,108 @@
+import kotlinx.coroutines.flow.fold
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.taktik.icure.asynclogic.MaintenanceTaskLogic
+import org.taktik.icure.asynclogic.impl.filter.Filters
+import org.taktik.icure.services.external.rest.v1.dto.MaintenanceTaskDto
+import org.taktik.icure.services.external.rest.v1.dto.embed.TaskStatusDto
+import org.taktik.icure.services.external.rest.v1.mapper.MaintenanceTaskMapper
+import org.taktik.icure.test.ICureTestApplication
+import org.taktik.icure.test.generateRandomString
+import org.taktik.icure.test.removeEntities
+import org.taktik.icure.services.external.rest.v1.dto.filter.maintenancetask.MaintenanceTaskAfterDateFilter
+
+@SpringBootTest(
+	classes = [ICureTestApplication::class],
+	properties = ["spring.main.allow-bean-definition-overriding=true"],
+	webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("app")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class MaintenanceTaskAfterDateFilterTest @Autowired constructor(
+	private val filters: Filters,
+	private val maintenanceTaskLogic: MaintenanceTaskLogic,
+	private val maintenanceTaskMapper: MaintenanceTaskMapper
+) {
+
+	private val testBatch = 10
+	private val currentTimestamp = System.currentTimeMillis()
+	private val alphabet: List<Char> = ('a'..'z').toList() + ('A'..'Z') + ('0'..'9')
+	private var taskIds: List<String> = listOf()
+
+	@BeforeAll
+	fun importMaintenanceTasks() {
+		runBlocking {
+			val taskIdsBefore = (1..testBatch).fold(listOf<String>()) { acc, _ ->
+				val taskId = generateRandomString(20, alphabet)
+				val newTask = MaintenanceTaskDto(
+					id = taskId,
+					created = currentTimestamp - nextInt(100, 10000),
+					status = TaskStatusDto.pending
+				)
+				maintenanceTaskLogic.createEntities(listOf(maintenanceTaskMapper.map(newTask))).collect()
+				acc + taskId
+			}
+			taskIds = (1..testBatch).fold(taskIdsBefore) { acc, _ ->
+				val taskId = generateRandomString(20, alphabet)
+				val newTask = MaintenanceTaskDto(
+					id = taskId,
+					created = currentTimestamp + nextInt(100, 10000),
+					status = TaskStatusDto.pending
+				)
+				maintenanceTaskLogic.createEntities(listOf(maintenanceTaskMapper.map(newTask))).collect()
+				acc + taskId
+			}
+		}
+	}
+
+	@Test
+	fun onlyTasksCreatedAfterTheDateSpecifiedInTheFilterAreReturned() {
+		runBlocking {
+			val dateFilter = MaintenanceTaskAfterDateFilter(date = currentTimestamp)
+			val filteredTasksCount = filters.resolve(dateFilter).fold(0) { acc, taskId ->
```

Comment:
To make the test more efficient, you can do it this way : 
```
val filteredTasksCount = filters.resolve(dateFilter).let { ids -> 
   getGenericDAO().getEntities(ids) 
}.onEach { task ->
   assertNotNull(task)
   assertNotNull(task!!.created)
   assert(task.created!! > currentTimestamp)
}.count()
```
Using getEntities() instead of getEntity for each task will allow you to do one request instead of 10. 
Also, this decouples the logic into your fold : First, in the onEach, you do your assertions, and afterwards you count the elements of your flow for your last filteredTasksCount assertion.